src/main/archetype/core/src/test/java/core/models/MyProductTeaserImplTest.java

```diff
@@ -0,0 +1,189 @@
+class MyProductTeaserImplTest {
+
+    private final static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private static final String PAGE = "/content/page";
+    private static final String PRODUCTTEASER_NO_BADGE = "productteaser-no-badge";
+    private static final String PRODUCTTEASER_BADGE_FALSE = "productteaser-badge-false";
+    private static final String PRODUCTTEASER_BADGE_TRUE_NO_AGE = "productteaser-badge-true-no-age";
+    private static final String PRODUCTTEASER_BADGE_TRUE_WITH_AGE = "productteaser-badge-true-with-age";
+
+    public final AemContext context = new AemContext(ResourceResolverType.JCR_MOCK);
+
+    private MyProductTeaser underTest;
+
+    private ProductTeaser productTeaser;
+
+    @Mock
+    private AbstractProductRetriever productRetriever;
+
+    @Mock
+    private ProductInterface product;
+
+    @BeforeEach
+    void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+        Mockito.when(productRetriever.fetchProduct()).thenReturn(product);
+        Mockito.when(product.getCreatedAt()).thenReturn("2020-01-01 00:00:00");
+
+        Page page = context.create().page(PAGE);
+        createResource(page, PRODUCTTEASER_NO_BADGE, null, null);
+        createResource(page, PRODUCTTEASER_BADGE_FALSE, false, null);
+        createResource(page, PRODUCTTEASER_BADGE_TRUE_NO_AGE, true, null);
+        createResource(page, PRODUCTTEASER_BADGE_TRUE_WITH_AGE, true, 3);
+
+        context.addModelsForClasses(MyProductTeaserImpl.class);
+    }
+
+    void createResource(Page page, String name, Object badge, Object age) {
+        Map<String, Object> props = new HashMap<>();
+        props.put("sling:resourceType", "venia/components/commerce/productteaser");
+        props.put("sling:resourceSuperType", "core/cif/components/commerce/productteaser/v1/productteaser");
+        if (badge != null) {
+            props.put("badge", badge);
+            if (age != null) {
+                props.put("age", age);
+            }
+        }
+        context.create().resource(page, name, props);
+    }
+
+    void setup(String resourceName) throws Exception {
+        Page page = context.currentPage(PAGE);
+        context.currentResource(PAGE + "/jcr:content/" +resourceName);
+        Resource teaserResource = context.resourceResolver().getResource(PAGE + "/jcr:content/" +resourceName);
+
+        // This sets the page attribute injected in the models with @Inject or @ScriptVariable
+        SlingBindings slingBindings = (SlingBindings) context.request().getAttribute(SlingBindings.class.getName());
+        slingBindings.setResource(teaserResource);
+        slingBindings.put(WCMBindingsConstants.NAME_CURRENT_PAGE, page);
+        slingBindings.put(WCMBindingsConstants.NAME_PROPERTIES, teaserResource.getValueMap());
+        try {
+            slingBindings.put("urlProvider", Mockito.mock(Class.forName( "com.adobe.cq.commerce.core.components.services.UrlProvider" )));
```

Comment:
We should remember to fix this after we release the new version of the components.